### ISSUE TITLE
History Issue in React Router 4

### ISSUE THREAD

**Issue by tresheffron** [mjrussell/redux-auth-wrapper#233]
I know I must be missing something easy here but am now stuck trying to setup my first app with redux-auth-wrapper. I was following the example project for RR4 and saw the Troubleshooting guide for getting around the the history undefined problem. As far as I can tell, the history object _should_ be getting passed along but I was getting the "Cannot read property 'replace' of undefined" error on "return history.replace;" in redirect.js. Then when I tried the withRouter workaround, I get "You should not use <Route> or withRouter() outside a `<Router>`."

```javascript
import React from 'react';
import { connect } from 'react-redux'
import { withRouter } from 'react-router';
import { BrowserRouter, Route, Redirect } from 'react-router-dom';

import { userIsAuthenticatedRedirect, userIsNotAuthenticatedRedirect } from './auth'
import { logout } from './actions/user'

import LoginComponent from './components/Login'
import AppComponent from './App';

// const Login = withRouter(userIsNotAuthenticatedRedirect(LoginComponent));
// const App = withRouter(userIsAuthenticatedRedirect(AppComponent));

const Login = userIsNotAuthenticatedRedirect(LoginComponent);
const App = userIsAuthenticatedRedirect(AppComponent);

function router ({ user, logout }) {
  return (
    <BrowserRouter>
      <div>
        <Route exact path="/" render={() => (
            <Redirect to="/app"/>
        )}/>
        <Route path="/login" component={Login}/>
        <Route path="/app" component={App} />
      </div>
    </BrowserRouter>
  );
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps, { logout })(App)
```

auth.js is a direct copy + paste of the auth.js file in the RR4 example.

My issue/question is: should I expect the connectedRouterRedirect (that is userIsAuthenticatedRedirect) to have access to the history object and I'm just looking in the wrong place to sort it out? Or is this expected with this setup, in which case, why is the withRouter workaround throwing an error?

Thank you.

**Comment by mjrussell** [comment]
Yes the `connectedRouterRedirect` does need access to the history object. I think your issue is that your `<div>` in the middle prevents passing the props down to the `Route` If you removed it I think you wouldn't have the issue. However, if you want to keep your div, you need to apply the `withRouter` inside the `Route` like `<Route path="/app" component={withRouter(App)} />`

**Comment by tresheffron** [comment]
Thank you for the insight and the suggestion. I just tried a different more nuclear option yesterday. I was trying to add the pattern into my existing code, and I'm sure there was something else I had wrong that influenced this (though maybe it was just the div). I scrapped my whole project, create a clean react app, dragged over the lock file, then pulled over the examples files as is. Made some slight modifications to make it work with react scripts, then fired it up and it worked. All of that points to something in how I setup the code on my side. So i'll close this as a user issue.
